src/components/main/menu/DateSelector/DateRangePicker.jsx

```diff
@@ -0,0 +1,218 @@
+            paddingLeft: '18px',
+            paddingTop: '18px',
+          }}
+        >
+          {title}
+        </p>
+        <button
+          type="button"
+          className="delete"
+          aria-label="close"
+          onClick={closeModal}
+        />
+      </header>
+
+      {/* ---------- Modal Card Body - main content ---------- */}
+      <section
+        className="modal-card-body"
+        style={{
+          overflow: 'visible',
+          display: 'flex',
+          alignItems: 'center',
+        }}
+      >
+        <div className="container">
+          <div className="container" style={containerDivStyle}>
+            &nbsp;
+            <span style={inputSpanStyle}>
+              Start Date
+            </span>
+            <DatePicker
+              todayButton="Today"
+              selected={startDate}
+              startDate={startDate}
+              endDate={endDate}
+              minDate={moment('01/01/2015', 'MM/DD/YYYY').toDate()}
+              maxDate={moment().toDate()}
+              selectsStart
+              showYearDropdown
+              showMonthDropdown
+              showPopperArrow={false}
+              popperPlacement="right"
+              onChange={(date) => handleDateChange(updateLocalStart, date)}
+              placeholderText="MM/DD/YYYY"
+            />
+          </div>
+          <div
+            className="container"
+            style={{
+              background: COLORS.FORMS.STROKE,
+              ...containerDivStyle,
+            }}
+          >
+            &nbsp;
+            <span style={{ ...inputSpanStyle }}>
+              End Date
+            </span>
+            <DatePicker
+              todayButton="Today"
+              selected={endDate}
+              startDate={startDate}
+              endDate={endDate}
+              minDate={startDate}
+              maxDate={new Date()}
+              showYearDropdown
+              showMonthDropdown
+              selectsStart
+              showPopperArrow={false}
+              popperPlacement="right"
+              onChange={(date) => handleDateChange(updateLocalEnd, date)}
+              placeholderText="MM/DD/YYYY"
+            />
+          </div>
+        </div>
+      </section>
+
+      {/* ---------- Modal Card Footer - button(s) ---------- */}
+      <footer
+        className="modal-card-foot"
+        style={{
+          background: headerStyle.background,
+          border: 'none',
+          textAlign: 'center',
+          borderRadius: '0',
+        }}
+      >
+        <div className="container" style={{ paddingBottom: '30px' }}>
+          <Button
+            id="date-range-button"
+            label="Save"
+            style={{
+              background: COLORS.BRAND.CTA1,
+              color: COLORS.FONTS,
+              fontWeight: 'bold',
+              width: '125px',
+              height: '42px',
+            }}
+            handleClick={() => {
+              if (startDate && endDate) {
```

Comment:
Can we do some more edge case validation here? cover cases where start/end might be a truthy value (i.e empty string)

Reply:
The reason I didn't really do any validation here is because the react date-picker library seemed to handle it all out of the box. The `Datepicker` controls the date input and calendar popup. If the value entered in the input can't be coerced into a valid date, react date-picker automatically nulls out in the input. I couldn't find a way to force an invalid input through. Props also control start/end dates to ensure dates only in valid range can be chosen. 

##### Input -> Result
1/20/15 -> 01/20/2015
11 -> 01/01/2020
01/01/01abc -> 01/01/2020
'' -> null
0 -> null
abcdef -> null
bunch of spaces -> null

Let me know if you still think some additional validation might be good. It's also likely there'll be some changes here when we decide what time format we want to use.

Reply:
poifect